### PR TITLE
Fix passwrd expiration issue in passw-hardening

### LINKED ISSUE #6628: During run passw-hardening TC DUT may become inaccessible
**Description**
When running `passw_hardening TC` on SONiC image which is older than expiration value that TC is setting (the lowest value for now is 10 days), DUT can become inaccessible. And when trying to ssh to DUT you will be asked to change passwd as it expired, this may have negative effect during regression runs. 
```
You are required to change your password immediately (password expired).
You are required to change your password immediately (password expired).

You are on
  ____   ___  _   _ _  ____
 / ___| / _ \| \ | (_)/ ___|
 \___ \| | | |  \| | | |
  ___) | |_| | |\  | | |___
 |____/ \___/|_| \_|_|\____|

-- Software for Open Networking in the Cloud --

Unauthorized access and/or use are prohibited.
All access and/or use are subject to monitoring.

Help:    https://sonic-net.github.io/SONiC/

WARNING: Your password has expired.
You must change your password now and login again!
```

**Steps to reproduce the issue:**
1. Deploy Sonic image which is 10 days old or more
2. Run` passw_hardening` TC
3.

**Describe the results you received:**
DUT became inaccessible, and requires to change password 

**Describe the results you expected:**
TC passed and DUT is accessible 

**Output of `show version`:**
```
SONiC Software Version: SONiC.master.159244-dirty-20221011.101352
Distribution: Debian 11.5
Kernel: 5.10.0-12-2-amd64
Build commit: cb707b796
Build date: Tue Oct 11 16:28:46 UTC 2022
Built by: AzDevOps@sonic-build-workers-0027YA
Platform: x86_64-arista_7170_64c
HwSKU: Arista-7170-64C
ASIC: barefoot
```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

### PR DESCRIPTION
**Pull request by AndriiLozovyi**
Signed-off-by: Andrii-Yosafat Lozovyi <andrii-yosafatx.lozovyi@intel.com>

### Description of PR

Summary: When running passw_hardening TC on SONiC image which is older than 10 days, DUT can become inaccessible.
This happens after `configuring passw-hardening expiration value of 10 days`, and if you're SONiC image is older than 10 days, you're password will be expired and you wont be able to ssh to DUT until password will be changed.
```
You are required to change your password immediately (password expired).
You are required to change your password immediately (password expired).

You are on
  ____   ___  _   _ _  ____
 / ___| / _ \| \ | (_)/ ___|
 \___ \| | | |  \| | | |
  ___) | |_| | |\  | | |___
 |____/ \___/|_| \_|_|\____|

-- Software for Open Networking in the Cloud --

Unauthorized access and/or use are prohibited.
All access and/or use are subject to monitoring.

Help:    https://sonic-net.github.io/SONiC/

WARNING: Your password has expired.
You must change your password now and login again!
```
This might have negative effect on regression runs. In order to fix this issue, we might check when last time password was changed, and then calculate whether expiration value test case are about to set is exceeding aging time of password.
Fixes #6628

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
Avoid DUT becoming inaccessible during passw-hardening run
#### How did you verify/test it?
Run passw_hardening on `SONiC.master.159244-dirty-20221011.101352`:
```
passw_hardening/test_passw_hardening.py::test_passw_hardening_en_dis_policies[2-dut] PASSED                                                                   
passw_hardening/test_passw_hardening.py::test_passw_hardening_history[2-dut] PASSED                                                                           
passw_hardening/test_passw_hardening.py::test_passw_hardening_age_expiration[2-dut] PASSED                                                                    
passw_hardening/test_passw_hardening.py::test_passw_hardening_age_expiration_warning[2-dut] PASSED                                                            
passw_hardening/test_passw_hardening.py::test_passw_hardening_len_min[2-dut] PASSED                                                                           
passw_hardening/test_passw_hardening.py::test_passw_hardening_policies_digits[2-dut] PASSED                                                                   
passw_hardening/test_passw_hardening.py::test_passw_hardening_policies_lower_class[2-dut] PASSED                                                              
passw_hardening/test_passw_hardening.py::test_passw_hardening_policies_upper_class[2-dut] PASSED                                                              
passw_hardening/test_passw_hardening.py::test_passw_hardening_policies_special_class[2-dut] PASSED                                                            
passw_hardening/test_passw_hardening.py::test_passw_hardening_policy_reject_user_passw_match[2-dut] PASSED 
```     
#### Any platform specific information?
```
SONiC Software Version: SONiC.master.159244-dirty-20221011.101352
Distribution: Debian 11.5
Kernel: 5.10.0-12-2-amd64
Build commit: cb707b796
Build date: Tue Oct 11 16:28:46 UTC 2022
Built by: AzDevOps@sonic-build-workers-0027YA
Platform: x86_64-arista_7170_64c
HwSKU: Arista-7170-64C
ASIC: barefoot
```

### CODE REVIEW

**Comment by AndriiLozovyi** [comment]
@wangxin, @yxieca
Could you please have a look ?

**Review by yxieca — APPROVED** [review]

File: tests/passw_hardening/test_passw_hardening.py [hunk 1]
```diff
@@ -88,6 +87,25 @@ def get_passw_expire_time_existing_user(duthost, normal_account):
...
+    Determine whether last password change date will be expired compared
+    to expiration value to be set
+    Args:
+        duthost: duthost object
+        expiration_value: expiration value to be checked
+    Return:
+        expiration: Expiration value to be set
+    """
+    msg = "Expiration diff is {}; Expected expiration to set {}; expiration used {}"
+    last_passw_change = duthost.shell('chage -l `echo "$USER"` | head -1')['stdout']
```
Comment (yxieca) [inline tests/passw_hardening/test_passw_hardening.py:101]: Is there a case that the date/time is not available?

Comment (AndriiLozovyi) [reply tests/passw_hardening/test_passw_hardening.py:101]: It doesn't look like there is such a case, Last password change date is present after adding new user already:
```
$ sudo useradd test_a
$ sudo chage -l test_a | head -1
Last password change                            : Oct 27, 2022
```

**Comment by AndriiLozovyi** [comment]
@yxieca Can this be merged? Thank you in advance!

**Pull request merged**